example/example.lisp

```diff
@@ -7,12 +7,29 @@
 ;; First define your API group, and its API functions.
 
 (eval-when (:compile-toplevel :load-toplevel :execute)
-  (define-api-group example))
+  (define-api-group example
+    ;; This documentation string will be included at the top of the .h
+    ;; file that is generated for this API-GROUP. Here, we show an
+    ;; example of a doxygen @file directive, but there is no requirement
+    ;; to use doxygen syntax. CALLBACK-HEAVEN simply treats these as an
+    ;; opaque string and emits them between C-style comment delimiters,
+    ;; beginning with /** and ending with */.
+    "@file example.h
```

Comment:
This is a little janky since the actual file name will depend on what the caller passes to `emit-api-files`, which means it's up to the caller to make sure it matches.

But I also didn't want to auto-emit the filename or something like that, in order to not bake-in doxygen semantics to the docstrings.  Another option would be to allow some kind of simple substitution in the docstrings, so that something like `%%FILENAME%%` gets replaced with the actual filename, but again I didn't want to get too carried away.

Reply:
KISS 💋